src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOptimizedNullableOperators.cs

```diff
@@ -3486,5 +3536,1585 @@ .maxstack  2
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/52629")]
+        public void NullableConstant_Short_E0()
+        {
+            var source = """
+                C.Run(0);
+                C.Run(1);
+                C.Run(2);
+                C.Run(null);
+
+                class C
+                {
+                    public static void Run(short? x)
+                    {
+                        System.Console.Write(M(x) ? 1 : 0);
+                    }
+                    static bool M(short? x) => x == (short)0;
+                }
+                """;
+            var output = "1000";
+            var il = """
+                {
+                  // Code size       56 (0x38)
+                  .maxstack  2
+                  .locals init (int? V_0,
+                                int V_1,
+                                short? V_2,
+                                int? V_3)
+                  IL_0000:  ldarg.0
+                  IL_0001:  stloc.2
+                  IL_0002:  ldloca.s   V_2
+                  IL_0004:  call       "bool short?.HasValue.get"
+                  IL_0009:  brtrue.s   IL_0016
+                  IL_000b:  ldloca.s   V_3
+                  IL_000d:  initobj    "int?"
+                  IL_0013:  ldloc.3
+                  IL_0014:  br.s       IL_0022
+                  IL_0016:  ldloca.s   V_2
+                  IL_0018:  call       "short short?.GetValueOrDefault()"
+                  IL_001d:  newobj     "int?..ctor(int)"
+                  IL_0022:  stloc.0
+                  IL_0023:  ldc.i4.0
+                  IL_0024:  stloc.1
+                  IL_0025:  ldloca.s   V_0
+                  IL_0027:  call       "int int?.GetValueOrDefault()"
+                  IL_002c:  ldloc.1
+                  IL_002d:  ceq
+                  IL_002f:  ldloca.s   V_0
+                  IL_0031:  call       "bool int?.HasValue.get"
+                  IL_0036:  and
+                  IL_0037:  ret
+                }
+                """;
+            var verifier = CompileAndVerify(source, options: TestOptions.ReleaseExe, expectedOutput: output);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.M", il);
+            verifier = CompileAndVerify(source, options: TestOptions.DebugExe, expectedOutput: output);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.M", il);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/52629")]
+        public void NullableConstant_Short_E1()
+        {
+            var source = """
+                C.Run(0);
+                C.Run(1);
+                C.Run(2);
+                C.Run(null);
+
+                class C
+                {
+                    public static void Run(short? x)
+                    {
+                        System.Console.Write(M(x) ? 1 : 0);
+                    }
+                    static bool M(short? x) => x == 1;
```

Comment:
>1 [](http://example.com/codeflow?start=52&length=1)

Should this be `(short)1`?

Reply:
That results in longer IL, you can see that in SharpLab. Same for other less-than-32bit ints. Doesn't seem to have any effect on the optimization as it's implented